clients/drcachesim/tracer/output.cpp

```diff
@@ -1004,14 +1007,33 @@ process_and_output_buffer(void *drcontext, bool skip_size_cap)
     }
 
     if (do_write) {
-        bool hit_window_end = false;
-        if (op_trace_for_instrs.get_value() > 0) {
+
+        if (op_L0_filter_until_instrs.get_value() && mode == BBDUP_MODE_L0_FILTER) {
+            uintptr_t toadd =
+                *(uintptr_t *)TLS_SLOT(data->seg_base, MEMTRACE_TLS_OFFS_ICOUNT);
+            bool reached_L0_filter_until_instrs_limit = count_traced_instrs(
+                drcontext, toadd, op_L0_filter_until_instrs.get_value());
+            if (reached_L0_filter_until_instrs_limit) {
+                data->bytes_written = 0;
```

Comment:
IIUC, for the multiple window case, control will reach here one time for each window. This means `bytes_written` will track #bytes only for the last window, not total bytes written yet. So `-max_trace_size` won't work as expected?

Reply:
Fixed this.